src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/Extensions/ServiceCollectionKeyedDescriptorExtensions.cs

```diff
@@ -0,0 +1,416 @@
+// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
Nit: I think the convention for partial classes is to call this something like `ServiceCollectionDescriptorExtensions.Keyed.cs`. It makes it easier to find when looking for "ServiceCollectionDescriptorExtensions".